### PR TITLE
Mesh_3: Fix std::atomic wrong function 

### LINKED ISSUE #5216: Tetrahedral remeshing does not compile with -DCGAL_LINKED_WITH_TBB
## Issue Details

Tetrahedral remeshing does not compile with `-DCGAL_LINKED_WITH_TBB`.

## Source Code

With this patch:
```patch
diff --git a/Tetrahedral_remeshing/examples/Tetrahedral_remeshing/CMakeLists.txt b/Tetrahedral_remeshing/examples/Tetrahedral_remeshing/CMakeLists.txt
index cc0ba60e6e2..f34cb525959 100644
--- a/Tetrahedral_remeshing/examples/Tetrahedral_remeshing/CMakeLists.txt
+++ b/Tetrahedral_remeshing/examples/Tetrahedral_remeshing/CMakeLists.txt
@@ -30,3 +30,4 @@ create_single_source_cgal_program(
   "mesh_and_remesh_polyhedral_domain_with_features.cpp")
 target_link_libraries(mesh_and_remesh_polyhedral_domain_with_features
                       PUBLIC CGAL::Eigen_support)
+target_compile_definitions(tetrahedral_remeshing_example PRIVATE CGAL_LINKED_WITH_TBB)
```
then the example `tetrahedral_remeshing_example` no longer compile:
```
[lrineau@bonnard]~/Git/cgal-master/build/examples/Tetrahedral_remeshing% make tetrahedral_remeshing_example
Building CXX object examples/Tetrahedral_remeshing/CMakeFiles/tetrahedral_remeshing_example.dir/tetrahedral_remeshing_example.cpp.o
In file included from /home/lrineau/Git/cgal-master/Mesh_3/include/CGAL/Mesh_cell_base_3.h:29,
                 from /home/lrineau/Git/cgal-master/Tetrahedral_remeshing/include/CGAL/Tetrahedral_remeshing/Remeshing_cell_base_3.h:18,
                 from /home/lrineau/Git/cgal-master/Tetrahedral_remeshing/include/CGAL/Tetrahedral_remeshing/Remeshing_triangulation_3.h:21,
                 from /home/lrineau/Git/cgal-master/Tetrahedral_remeshing/examples/Tetrahedral_remeshing/tetrahedral_remeshing_example.cpp:3:
/home/lrineau/Git/cgal-master/Mesh_3/include/CGAL/Mesh_3/Mesh_surface_cell_base_3.h: In member function ‘void CGAL::Mesh_3::Mesh_surface_cell_base_3_base<CGAL::Parallel_tag>::set_facet_visited(int)’:
/home/lrineau/Git/cgal-master/Mesh_3/include/CGAL/Mesh_3/Mesh_surface_cell_base_3.h:100:18: error: ‘struct std::atomic<char>’ has no member named ‘compare_and_swap’
  100 |     while (bits_.compare_and_swap(current_bits | (1 << facet), current_bits) != current_bits)
      |                  ^~~~~~~~~~~~~~~~
/home/lrineau/Git/cgal-master/Mesh_3/include/CGAL/Mesh_3/Mesh_surface_cell_base_3.h: In member function ‘void CGAL::Mesh_3::Mesh_surface_cell_base_3_base<CGAL::Parallel_tag>::reset_visited(int)’:
/home/lrineau/Git/cgal-master/Mesh_3/include/CGAL/Mesh_3/Mesh_surface_cell_base_3.h:111:18: error: ‘struct std::atomic<char>’ has no member named ‘compare_and_swap’
  111 |     while (bits_.compare_and_swap(current_bits & (15 & ~(1 << facet)), current_bits) != current_bits)
      |                  ^~~~~~~~~~~~~~~~
make[3]: *** [examples/Tetrahedral_remeshing/CMakeFiles/tetrahedral_remeshing_example.dir/build.make:82: examples/Tetrahedral_remeshing/CMakeFiles/tetrahedral_remeshing_example.dir/tetrahedral_remeshing_example.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:34633: examples/Tetrahedral_remeshing/CMakeFiles/tetrahedral_remeshing_example.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:34640: examples/Tetrahedral_remeshing/CMakeFiles/tetrahedral_remeshing_example.dir/rule] Error 2
make: *** [Makefile:251: examples/Tetrahedral_remeshing/CMakeFiles/tetrahedral_remeshing_example.dir/rule] Error 2
```

`Mesh_3/include/CGAL/Mesh_3/Mesh_surface_cell_base_3.h` is wrong since commit  85712ba28c63b708f106517a78229665d9a3ef23 from PR #4468. cc: @maxGimeno 

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux Fedora 32, but certainly cross-platform

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Fix `std::atomic::compare_and_swap()` that doesn't wxist into `compare_exchange_weak()`

## Release Management

* Affected package(s):Tetrahedral_remeshing, Mesh_3
* Issue(s) solved (if any): fix #5216

### CODE REVIEW

File: Tetrahedral_remeshing/examples/Tetrahedral_remeshing/CMakeLists.txt [hunk 1]
```diff
@@ -32,3 +34,6 @@ create_single_source_cgal_program( "tetrahedral_remeshing_from_mesh.cpp")
 
 create_single_source_cgal_program( "mesh_and_remesh_polyhedral_domain_with_features.cpp" )
 target_link_libraries(mesh_and_remesh_polyhedral_domain_with_features PUBLIC CGAL::Eigen_support)
+if(TARGET CGAL::TBB_support)
+target_link_libraries(mesh_and_remesh_polyhedral_domain_with_features PRIVATE CGAL::TBB_support)
+endif()
```
Comment (lrineau) [inline Tetrahedral_remeshing/examples/Tetrahedral_remeshing/CMakeLists.txt:39]:
```suggestion
if(TARGET CGAL::TBB_support)
  target_link_libraries(mesh_and_remesh_polyhedral_domain_with_features PRIVATE CGAL::TBB_support)
endif()
```

File: Mesh_3/include/CGAL/Mesh_3/Mesh_surface_cell_base_3.h [hunk 2]
```diff
@@ -95,9 +95,10 @@ class Mesh_surface_cell_base_3_base<Parallel_tag>
   /// Marks \c facet as visited
   void set_facet_visited (const int facet)
   {
+    std::cerr<<"coucou"<<std::endl;
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_3/Mesh_surface_cell_base_3.h:98]:
```suggestion
```

Comment (maxGimeno) [reply Mesh_3/include/CGAL/Mesh_3/Mesh_surface_cell_base_3.h:98]: Oh !!!

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-13.shtml

**Pull request merged**